docs/data/material/guides/right-to-left/right-to-left.md

```diff
@@ -1,37 +1,47 @@
-# Right-to-left
+# Right-to-left language support
 
-<p class="description">Right-to-left languages such as Arabic, Persian, or Hebrew are supported. To change the direction of Material UI components you must follow the following steps.</p>
+<p class="description">Learn how to implement right-to-left (RTL) text with Material UI to support languages such as Arabic, Persian, and Hebrew.</p>
 
-## Steps
+This guide outlines the three steps necessary to change the direction of text-based components in Material UI to support RTL languages, as shown in the demo below:
 
-### 1. HTML
+{{"demo": "RTLDemo.js"}}
 
-Make sure the `dir` attribute is set on the `html` tag, otherwise native components will break:
+## 1. Set the HTML direction
+
+### Globally
+
+Add `dir="rtl"` to the app's root `<html>` to set the global text direction:
 
 ```html
 <html dir="rtl"></html>
 ```
 
-If you need to change the direction of the text at runtime, but React does not control the root HTML element, you may use the JS API:
+If your React app doesn't control the root `<html>` and you need to change the text direction at runtime, use the JavaScript API instead:
 
 ```js
 document.dir = 'rtl';
 ```
 
-As an alternative to the above, you can also wrap your application (or part of it) in an element with the `dir` attribute.
-This, however, will not work correctly with portaled elements, such as Dialogs, as they will render outside of the element with the `dir` attribute.
+### Locally
+
+Add the `dir="rtl"` attribute to any other HTML element or React component if you need limit the scope of the text direction to that element and its children.
 
-To fix the portaled components, add an explicit `dir` attribute to them:
+:::warning
+This won't work with portaled elements, such as Dialogs, as they render outside of the element with the `dir` attribute.
+
+To fix it, make sure to add the `dir` attribute directly to them:
 
 ```jsx
 <Dialog dir="rtl">
   <MyComponent />
 </Dialog>
```

Comment:
```suggestion
Components that use React portals (like the [Dialog](/material-ui/react-dialog)) do _not_ inherit the `dir` attribute from parents, because they actually render outside of their parental DOM tree.

You must apply the `dir` attribute directly to these components if it's not [globally](#globally) defined as right-to-left:

```jsx
<Box dir="rtl">
  <Dialog /> // ❌ this Dialog will still be left-to-right (the default)
</Box>

<Box dir="rtl">
  <Dialog dir="rtl" /> // ✅ this Dialog will be right-to-left as intended
</Box>
```
```